### PR TITLE
12 cicd pipelines

### LINKED ISSUE #12: CI/CD pipelines
It would be nice if we can make this SPA use CI/CD to re-build the `.json` files whenever we merge an updated `.csv` file into the codebase.

- [ ] on merge validate the code
- [ ] use an action job to lint the code
- [ ] use an action job to run the `.py` script to build a new `.json`
- [ ] use an action job to scp the SPA to our server
- [ ] use github secrets to store the environment variables

### PR DESCRIPTION
**Pull request by mrozanoff**
### Description: ✍️
*Created pipeline for JSON file to be created automatically on push

### Issue(s): 🎟️
* #12

### CODE REVIEW

File: .github/workflows/update_json.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: Update Fungi Taxonomy JSON
+
+# Run this workflow on a push to main or on a schedule
+on:
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: '0 0 * * *'  # Runs daily at midnight; adjust as needed
```
Comment (Elaniobro) [inline .github/workflows/update_json.yml]: This is dope, but I doubt it needs to run nightly. How often do you think it _should_ run, on top of a push? 

Monthly seems more reasonable. What do you think?

File: .github/workflows/update_json.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        run: python create_json_taxonomy.py
+
+      # Commit and push changes if fungi_taxonomy.json was modified
+      - name: Commit and push changes
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          if [ -n "$(git status --porcelain fungi_taxonomy.json)" ]; then
+            git add fungi_taxonomy.json
+            git commit -m "Update fungi_taxonomy.json"
```
Comment (Elaniobro) [inline .github/workflows/update_json.yml]: This commit should read:

`ci: update fungi_taxonomy.json`

**Review by Elaniobro — CHANGES_REQUESTED** [review]
This is great. Commit line needs a tweak and would like your opinion on cadence for the job to run

File: .github/workflows/update_json.yml (hunk 1 above)
Comment (mrozanoff) [reply .github/workflows/update_json.yml]: oh i forgot about this while working on the others parts, fixed

**Comment by mrozanoff** [comment]
monthly is fine, we can always change it

**Review by Elaniobro — APPROVED** [review]
:shipit:

**Pull request merged**